### PR TITLE
Fix if entry is a function

### PR DESCRIPTION
**Pull request by chvin**
If the entry is a function, the original logic will be treated as an object, causing the user's entry to be empty.
```js
// problem code
if (originalEntry === Object(originalEntry)) {
```

### CODE REVIEW

**Comment by adierkens** [comment]
Thanks for the contribution. 

Looks like the PR builds started throwing an error, for now I removed the offending line in https://github.com/adierkens/webpack-inject-plugin/pull/32. Can you rebase this PR?

Once it successfully builds on Circle I can merge/release it

**Comment by chvin** [comment]
Using ts @^3.1.3 can't be built on the master, ts@~3.1.3 can, I fixed.

File: src/main.ts [hunk 1]
```diff
@@ -93,7 +93,15 @@ export function injectEntry(
     return injectToArray(originalEntry, newEntry, options.entryOrder);
   }
 
-  if (originalEntry === Object(originalEntry)) {
+  if (typeof originalEntry === 'function') {
```
Comment (adierkens) [inline src/main.ts]: Can you add tests for this?

File: src/main.ts [hunk 2]
```diff
@@ -93,7 +93,15 @@ export function injectEntry(
...
-  if (originalEntry === Object(originalEntry)) {
+  if (typeof originalEntry === 'function') {
+    const callbackOriginEntry = originalEntry();
+    if (callbackOriginEntry instanceof Promise) { // can't handle Promise
+      return originalEntry;
+    }
+    return injectEntry(callbackOriginEntry, newEntry, options);
+  }
+
+  if (Object.prototype.toString.call(originalEntry).slice(8, -1) === 'Object') {
```
Comment (adierkens) [inline src/main.ts]: Since you added the above check for a function, the original object check should still be valid.

**Comment by adierkens** [comment]
Also, can you rebase to run the circle builds?

File: src/main.ts (hunk 1 above)
Comment (chvin) [reply src/main.ts]: > Can you add tests for this?

In the latest commit I added a test for the function entry.

File: src/main.ts (hunk 2 above)
Comment (chvin) [reply src/main.ts]: I think the original object check still be valid.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString

**Comment by chvin** [comment]
> Also, can you rebase to run the circle builds?

Sorry, I don't understand what circle builds is, how to work :(

**Comment by chvin** [comment]
image
It doesn't seem to matter to my PR.

**Comment by adierkens** [comment]
_Circle_ is a build server that builds and tests every pull request. 

I fixed the issue you saw (the reason it's currently failing) in https://github.com/adierkens/webpack-inject-plugin/pull/32, if you update your branch to include the latest changes on _master_, you'll get that fix and your builds and tests will be ran

**Pull request merged**